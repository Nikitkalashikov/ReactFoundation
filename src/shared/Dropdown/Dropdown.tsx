import React from "react";
import styles from './dropdown.scss';

interface IDropdownProps {
    button: React.ReactNode;
    children: React.ReactNode;
    isOpen?: boolean;
    onOpen?: () => void;
    onClose?: () => void;
    listClass?: string;
}

const NOOP = () => {};

export function Dropdown({ button, children, isOpen, onClose = NOOP, onOpen = NOOP, listClass}: IDropdownProps){
    const [isDropdownOpen, setIsDropdownOpen] = React.useState(isOpen);  
    React.useEffect(() => setIsDropdownOpen(isOpen), [isOpen]);
    React.useEffect(() => isDropdownOpen ? onOpen() : onClose(), [isDropdownOpen]);

    const handlerOpen = () => {
        if (isOpen === undefined){
            setIsDropdownOpen(!isDropdownOpen)
        }
    }

    return (
        <div className={styles.dropdown}>
            <div className={styles.dropdown__button} onClick={handlerOpen}>
                {button}
            </div>
            {isDropdownOpen && (
                <div className={styles.dropdown__list + ' ' + listClass} onClick={()=>setIsDropdownOpen(false)}>
                    {children}
                </div>
            )}
        </div>
    );
}