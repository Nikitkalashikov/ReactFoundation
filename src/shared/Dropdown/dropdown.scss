.dropdown{
    position: relative;
    width: 100%;

    &__button{
        width: auto;
    }

    &__item{
        color: var(--gray99);
        font-size: 14px;
        line-height: 18px;
    }

    &__list{
        position: absolute;
        top: calc( 100% + 19px);
        left: 50%;
        -webkit-transform: translateX(-50%);
        transform: translateX(-50%);
        width: max-content;
        height: max-content;
        background-color: var(--white);
        border-radius: 3px;
        overflow: hidden;
        box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.15);
        z-index: 10;
    }
}