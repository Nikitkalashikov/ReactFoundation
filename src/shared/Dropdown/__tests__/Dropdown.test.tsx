import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@zarconontol/enzyme-adapter-react-18';
import { Dropdown } from '../Dropdown';

configure({adapter: new Adapter()});

describe('Dropdown', () => {
    // test('should render', () => {
    //     const wrapper = shallow(<Dropdown children={<div />} button={<button />}/>)
    //     expect(wrapper).toBeDefined();
    //     expect(wrapper.find('div.dropdown').isEmptyRender()).toBeFalsy();
    // })

    test('should render (snapshot)', () => {
        const wrapper = shallow(<Dropdown children={<div />} button={<button />}/>)

        expect(wrapper).toMatchSnapshot()
    })
})