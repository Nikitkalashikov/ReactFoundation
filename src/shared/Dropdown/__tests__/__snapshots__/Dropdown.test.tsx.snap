// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render (snapshot) 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown__button"
    onClick={[Function]}
  >
    <button />
  </div>
</div>
`;
